### ISSUE TITLE
Bug: `func-names` schema allows extra elements

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#20134]
### Environment

Node version: v22.14.0
npm version: 10.9.2
Local ESLint version: v9.35.0
Global ESLint version: no
Operating System: windows

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [{]];
```
</details>

```js
/*eslint func-names: ["error", "always", { "generators": "never" }, "foo"]*/
```

### What did you expect to happen?

An error saying that the configuration for `func-names` is invalid due to an extra element `"foo"`. 

### What actually happened?

No errors.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgZnVuYy1uYW1lczogW1wiZXJyb3JcIiwgXCJhbHdheXNcIiwgeyBcImdlbmVyYXRvcnNcIjogXCJuZXZlclwiIH0sIFwiZm9vXCJdKi8iLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Participation

- [ ] I am willing to submit a pull request for this issue.

There's already a pull request: https://github.com/eslint/eslint/pull/20119

### Additional comments

Per our policies, this is a breaking change (see https://github.com/eslint/eslint/pull/20119#discussion_r2348159722).

**Comment by mdjermanovic** [comment]
This change has been accepted for ESLint v10.

**Comment by Revanwjy-alt** [comment]
main disini berhadiah menarik

jo777.help
